### ISSUE TITLE
fatal error overflow of count when invoking Observable.range

### ISSUE THREAD

**Issue by gotev** [ReactiveX/RxSwift#1870]
**Short description of the issue**:

fatal error `overflow of count` [here](https://github.com/ReactiveX/RxSwift/blob/master/RxSwift/Observables/Range.swift#L36) when invoking 

```swift
let count = 0
Observable.range(start: 0, count: count)
```

**Expected outcome**:

Empty Observable, to match the functional behavior of Swift Range:

```swift
let count = 0
(0..count).forEach { print("\($0)") }
```

In that case it prints nothing. Furthermore, RxJava also returns an empty Observable (Flowable in RxJava2):

```java
@CheckReturnValue
@BackpressureSupport(BackpressureKind.FULL)
@SchedulerSupport(SchedulerSupport.NONE)
public static Flowable<Integer> range(int start, int count) {
    if (count < 0) {
        throw new IllegalArgumentException("count >= 0 required but it was " + count);
    } else
    if (count == 0) {
        return empty();
    } else
    if (count == 1) {
        return just(start);
    } else
    if ((long)start + (count - 1) > Integer.MAX_VALUE) {
        throw new IllegalArgumentException("Integer overflow");
    }
    return RxJavaPlugins.onAssembly(new FlowableRange(start, count));
}
```

**What actually happens**:

fatal error `overflow of count` 

**Self contained code example that reproduces the issue**:

```swift
func myFunc(count: Int) -> Observable<[MyType]> {
    return Observable.range(start: 0, count: count)
        .concatMap { index in
            return //observable API call
        }.toArray()
}
```

**Platform/Environment**

  - [x] iOS
  - [x] macOS
  - [x] tvOS
  - [x] watchOS
  - [x] playgrounds

**How easy is to reproduce? (chances of successful reproduce after running the self contained code)**

  - [x] easy, 100% repro
  - [ ] sometimes, 10%-100%
  - [ ] hard, 2% - 10%
  - [ ] extremely hard, %0 - 2%

**Xcode version**:

```
10.1
```

**Installation method**:
  - [ ] CocoaPods
  - [x] Carthage
  - [ ] Git submodules

**I have multiple versions of Xcode installed**:
  (so we can know if this is a potential cause of your issue)
  - [ ] yes (which ones)
  - [x] no

**Level of RxSwift knowledge**:
  (this is so we can understand your level of knowledge
    and formulate the response in an appropriate manner)
  - [ ] just starting
  - [ ] I have a small code base
  - [x] I have a significant code base

**Comment by kzaher** [comment]
Hi @gotev ,

yeah, we should accept 0 as a valid value.

**Comment by freak4pc** [comment]
Made a quick PR to fix this #1871

**Comment by gotev** [comment]
Wow that's great!! Swift community at its best <3
